### ISSUE TITLE
Reverted to old version of project?

### ISSUE THREAD

**Issue by jhmacdon** [arunoda/meteor-up-legacy#319]
As soon as I updated mup, it reverted my project to a relatively old version...probably 4 or 5 deploys ago, maybe a handful of commits. Could the update have caused this?

**Comment by arunoda** [comment]
Hmm. Mup never changes git or local file system state.

May be your app gets reverted to an older state. Check sever logs.

**Comment by bmx269** [comment]
I have the same issue. Just happened.

**Comment by jbrozena22** [comment]
I'm noticing something similar. Is it possible that mup deploy doesn't clean out old processes before starting a new one? Because it looks like in my app some users are using code from a few (1-5) versions back.

**Comment by jhmacdon** [comment]
I noticed that it was trying to seed a ton of info during deployment, and if it didnt finish seeding then this stuff started happening. Watch logs with <code>mup logs -f</code>, add some <code>console.log</code> statements to your Meteor code and watch where your app ends up when it finishes testing for the apps state. 

Once this issue happened, I wasn't able to fix without destroying my droplet (Digital Ocean) and starting a new one. The actual fix is to run large seeds and such from some sort of admin panel (Houston is a good one).

**Comment by tehfailsafe** [comment]
Same. Client meeting in 1 hour and now it's displaying version from several commits back. mup setup mup reconfig power cycle server nothing fixes it. I was seeding data with fixtures.js and even after removing the file it keeps running the fixtures...

**Comment by 255kb** [comment]
Same problem here, two apps running behind nginx since months on the same VPS, just updated one (with simple git pull origin master and sudo mup deploy, as usual) and now the server is serving very old versions for both apps (like months old...). 
Restarting the server seems to serve the right version again, and then some minutes after the same old version got served.
I cannot figure out where this come from.

**Comment by hsduk** [comment]
I just experienced this problem. I don't think the error message is correct - "Redeploying previous version of the app".
My app grew in size over several deployments and as expected, the app took longer to start. 
I changed mup.json "deployCheckWaitTime" when I first deployed to 30 secs. Changing it to 45 secs removed this error message,
Would suggest that mup calculates the time to start the server and stores this.
Also change "deployCheckWaitTime" to start its clock when the last server startup time is up.
So if lastStartDuration (or average?) was 20seconds, having a deployCheckWaitTime of 15secs would trigger this error, that the deployment was unsuccessful after 35seconds ( lastStartDuration + deployCheckWaitTime ).
Setting deployCheckWaitTime to something bigger like 60seconds without polling the http port makes deployment take too long.
